### ISSUE TITLE
Reduce the overall time the `test race(xx)` Github Actions take

### ISSUE THREAD

**Issue by dwedul-figure** [provenance-io/provenance#826]
## Summary

It'd be nice if all the `test race(xx)` jobs took about the same time.

## Problem Definition

Currently, the `test race(01)` job takes more than twice as long as the other `test race(xx)` jobs. When trying to auto-merge several PRs, that job is usually the one we end up waiting on. Reducing its time would allow us to merge PRs faster (e.g. when we have 5 Dependabot PRs).

This is all defined in `.github/workflows/test.yml`. A list of all tests to run is created, then the `split` is used to separate that into 4 sub-lists. Each is then run simultaneously for both the `test (xx)` jobs and `test race (xx)` jobs.

As you can see in https://github.com/provenance-io/provenance/actions/runs/2333640892, the `test race(01)` job takes 9m48s, and the next longest one takes 4m47s.

## Proposal

I'm guessing that it isn't a single test that causes that longevity, and instead is a few slightly longer tests that all end up together in `01`.

Do some research to try to identify the reason `test race(01)` takes so long.

If it's one single test taking 4 minutes or more, split it out into a 5th file and add it to the matrices.

If it's several tests that just take a little longer, devise a way to spread out all the long-running tests. This might be as easy as devising a deterministic way to "randomize" the test list before calling `split`. Another option is to identify all the long-running tests, extract them from the `pkgs.txt` file into their own file. Then split `pkgs.txt` and that file separately and merge the corresponding results back into the 4 lists used.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by dwedul-figure** [comment]
Here's where the splitting happens: https://github.com/provenance-io/provenance/blob/ce72d3f4922d45e78d481443a7620c2096655435/.github/workflows/test.yml#L42-L43

Looking at the "test & coverage report creation" job in here: https://github.com/provenance-io/provenance/runs/6457227129?check_suite_focus=true it looks like `github.com/provenance-io/provenance/x/marker/simulation	371.502s` is the culprit.

So we'll want to just create a 5th parallel job just for that:
1. Update the splitting as follows: Put `github.com/provenance-io/provenance/x/marker/simulation` into a file named `pkgs.txt.part.sim`. Remove `github.com/provenance-io/provenance/x/marker/simulation` from `pkgs.txt` before calling `split` on it (maybe `grep -vF 'github.com/provenance-io/provenance/x/marker/simulation' pkgs.txt | split ...`?).
2. Add a step in `setup-tests` that uploads the `pkgs.txt.part.sim` artifact (like the others).
3. Add `"sim"` to the matrices of the `tests` and `test-race` jobs.
4. Update the `upload-coverage-report` job to also download the `"${{ needs.setup-tests.outputs.file-prefix }}-sim-coverage"` artifact.

**Comment by dwedul-figure** [comment]
The required checks only apply to 00 through 03. That is, when I added a `mrkrsim` entry to the matrix, it wasn't a required check.

It's weird. The `x/marker/simulation` tests take about 20 seconds as a regular test, but about 6 minutes as a race test. 
So I ended up making it the entirety of 03, and the rest of the tests are split among 00, 01, 02, 03.

The next longest is 2.5 minutes. So if we want to speed the required checks up even more, we'll need to specifically make changes to the `x/marker/simulation` tests.
